src/System.ComponentModel.TypeConverter/tests/Drawing/ImageFormatConverterTests.cs

```diff
@@ -0,0 +1,231 @@
+            Assert.True(_imgFmtConvFrmTD.CanConvertTo(null, typeof(string)), "TD string");
+            Assert.False(_imgFmtConvFrmTD.CanConvertTo(null, typeof(ImageFormat)), "TD ImageFormat");
+            Assert.False(_imgFmtConvFrmTD.CanConvertTo(null, typeof(Guid)), "TD Guid");
+            Assert.False(_imgFmtConvFrmTD.CanConvertTo(null, typeof(object)), "TD object");
+            Assert.False(_imgFmtConvFrmTD.CanConvertTo(null, typeof(int)), "TD int");
+        }
+
+        [ConditionalFact(Helpers.GdiplusIsAvailable)]
+        public void TestConvertFrom_ImageFormatToString()
+        {
+            Assert.Equal(_imageFmt, (ImageFormat)_imgFmtConv.ConvertFrom(null, CultureInfo.InvariantCulture, ImageFormat.Bmp.ToString()));
+            Assert.Equal(_imageFmt, (ImageFormat)_imgFmtConvFrmTD.ConvertFrom(null, CultureInfo.InvariantCulture, ImageFormat.Bmp.ToString()));
+        }
+
+        [ConditionalFact(Helpers.GdiplusIsAvailable)]
+        public void TestConvertFrom_ThrowsNotSupportedException()
+        {
+            Assert.Throws<NotSupportedException>(() => _imgFmtConv.ConvertFrom("System.Drawing.String"));
+            Assert.Throws<NotSupportedException>(() => _imgFmtConv.ConvertFrom(null, CultureInfo.InvariantCulture, "System.Drawing.String"));
+            Assert.Throws<NotSupportedException>(() => _imgFmtConv.ConvertFrom(null, CultureInfo.InvariantCulture, ImageFormat.Bmp));
+            Assert.Throws<NotSupportedException>(() => _imgFmtConv.ConvertFrom(null, CultureInfo.InvariantCulture, ImageFormat.Bmp.Guid));
+            Assert.Throws<NotSupportedException>(() => _imgFmtConv.ConvertFrom(null, CultureInfo.InvariantCulture, new object()));
+            Assert.Throws<NotSupportedException>(() => _imgFmtConv.ConvertFrom(null, CultureInfo.InvariantCulture, 10));
+
+            Assert.Throws<NotSupportedException>(() => _imgFmtConvFrmTD.ConvertFrom("System.Drawing.String"));
+            Assert.Throws<NotSupportedException>(() => _imgFmtConvFrmTD.ConvertFrom(null, CultureInfo.InvariantCulture, "System.Drawing.String"));
+            Assert.Throws<NotSupportedException>(() => _imgFmtConvFrmTD.ConvertFrom(null, CultureInfo.InvariantCulture, ImageFormat.Bmp));
+            Assert.Throws<NotSupportedException>(() => _imgFmtConvFrmTD.ConvertFrom(null, CultureInfo.InvariantCulture, ImageFormat.Bmp.Guid));
+            Assert.Throws<NotSupportedException>(() => _imgFmtConvFrmTD.ConvertFrom(null, CultureInfo.InvariantCulture, new object()));
+            Assert.Throws<NotSupportedException>(() => _imgFmtConvFrmTD.ConvertFrom(null, CultureInfo.InvariantCulture, 10));
+        }
+
+        private ImageFormat ShortName(string imgFormatValue)
+        {
+            return (ImageFormat)_imgFmtConvFrmTD.ConvertFrom(null, CultureInfo.InvariantCulture, imgFormatValue);
+        }
+
+        [ConditionalFact(Helpers.GdiplusIsAvailable)]
+        public void ConvertFrom_ShortName()
```

Comment:
We should add a test where we convert from a ImageFormat with an unknown guid -- once you add the hability to convert from that.

Reply:
I will add the test. But I found out, that when I trying to convert from invalid string, `NotSupportedException` is thrown with the text:

> ImageFormatConverter cannot convert from System.String.

This is not true. It can convert from `System.String` in general, but it cannot convert from some invalid string. Are we going to address this somehow? Change the exception message for example?

Reply:
We had this issue here: https://github.com/dotnet/corefx/pull/28516#discussion_r202467189

So I would suggest instead throwing an `ArgumentException` or `FormatException` in that case because that means the format of the string is invalid.